Source/GSThemeDrawing.m

```diff
@@ -3475,15 +3496,343 @@ - (void) drawTableViewRow: (NSInteger)rowIndex
+              image = [NSImage imageNamed: @"common_ArrowRightH"];
+            }
+
+          if (![outlineView isExpandable: item])
+            {
+              image = AUTORELEASE([[NSImage alloc] initWithSize: NSMakeSize(14.0,14.0)]);
+            }
+
+          level = [outlineView levelForItem: item];
+          indentationFactor = indentationPerLevel * level;
+          imageCell = [[NSCell alloc] initImageCell: image];
+          imageRect = [outlineView frameOfOutlineCellAtRow: rowIndex];
+
+          if ([delegate respondsToSelector: @selector(outlineView:willDisplayOutlineCell:forTableColumn:item:)])
+            {
+              [delegate outlineView: outlineView
+			willDisplayOutlineCell: imageCell
+		     forTableColumn: tb
+			       item: item];
+            }
+
+          /* Do not indent if the delegate set the image to nil. */
+          if ([imageCell image])
+            {
+              imageRect.size.width = [image size].width;
+              imageRect.size.height = [image size].height;
+              [imageCell drawWithFrame: imageRect inView: outlineView];
+              drawingRect.origin.x
+                += indentationFactor + imageRect.size.width + 5;
+              drawingRect.size.width
+                -= indentationFactor + imageRect.size.width + 5;
+            }
+          else
+            {
+              drawingRect.origin.x += indentationFactor;
+              drawingRect.size.width -= indentationFactor;
+            }
+
+          RELEASE(imageCell);
+        }
+
+      [cell drawWithFrame: drawingRect inView: outlineView];
+      if (i == editedColumn && rowIndex == editedRow)
+        {
+          [cell _setInEditing: NO];
+          [cell setShowsFirstResponder: NO];
+        }
+    }  
+}
+
+- (void) drawCellViewRow: (NSInteger)rowIndex
+		clipRect: (NSRect)clipRect
+		  inView: (id)v
+{
+  NSInteger numberOfColumns = [v numberOfColumns];
+  CGFloat *columnOrigins = [v _columnOrigins];
+  NSArray *tableColumns = [v tableColumns];
+  NSInteger numberOfRows = [v numberOfRows];
+  NSInteger startingColumn; 
+  NSInteger endingColumn;
+  NSInteger i;
+  CGFloat x_pos;
+  id dataSource = [v dataSource];
+  id delegate = [v delegate];
+  BOOL hasMethod = NO;
+  NSTableColumn *outlineTableColumn = nil;
+
+  // If we have no data source, there is nothing to do...
+  if (dataSource == nil)
+    {
+      return;
+    }
+
+  // If the rowIndex is greater than the numberOfRows, done...
+  if (rowIndex >= numberOfRows)
+    {
+      return;
+    }
+  
+  // Check the delegate method...
+  if ([v isKindOfClass: [NSOutlineView class]])
+    {
+      outlineTableColumn = [v outlineTableColumn];
+      hasMethod = [delegate respondsToSelector: @selector(outlineView:viewForTableColumn:item:)];
+    }
+  else if ([v isKindOfClass: [NSTableView class]])
+    {
+      hasMethod = [delegate respondsToSelector: @selector(tableView:viewForTableColumn:row:)];
+    }
+  
+  /* Determine starting column as fast as possible */
```

Comment:
This seems to duplicate some of the code from the previous method. Maybe this could be resolved by splitting the methods differently and passing in more parameters?